README.md

```diff
@@ -482,145 +687,164 @@ function onThingHappended(cb){
 
-    ```javascript
-    var items
-      , messages
-      , length
-      , i
+    // good
+    const errorMessage = `This is a super long error that
+      was thrown because of Batman.
+      When you stop to think about
+      how Batman had anything to do
+      with this, you would get nowhere
+      fast.`
+    ```
 
-    messages = [{
-      state: 'success'
-      , message: 'This one worked.'
-    },{
-      state: 'success'
-      , message: 'This one worked as well.'
-    },{
-        state: 'error',
-        , message: 'This one did not work.'
-    }]
+   - When programmatically building up strings, use template strings instead of concatenation.
 
-    length = messages.length
+  > Why? Template strings give you a readable, concise syntax with proper newlines and string interpolation features.
 
+    ```javascript
     // bad
-    function inbox(messages){
-      items = '<ul>'
-
-      for (i = 0; i < length; i++){
-        items += '<li>' + messages[i].message + '</li>'
-      }
+    function sayHi(name) {
+      return 'How are you, ' + name + '?'
+    }
 
-      return items + '</ul>'
+    // bad
+    function sayHi(name) {
+      return ['How are you, ', name, '?'].join()
     }
 
     // good
-    function inbox(messages){
-      items = []
-
-      for (i = 0; i < length; i++){
-        items[i] = messages[i].message
-      }
-
-      return '<ul><li>' + items.join('</li><li>') + '</li></ul>'
+    function sayHi(name) {
+      return `How are you, ${name}?`
     }
     ```
 
 
-
 ### Functions
 
-  - Function expressions:
+- Use function declarations instead of function expressions.
+
+  > Why? Function declarations are named, so they're easier to identify in call stacks. Function declarations that aren't assigned to a variable are hoisted and can lead to expected bugs.
 
     ```javascript
-    // anonymous function expression
-    var anonymous = function(){
-      return true
-    }
+    // bad
+    const foo = function () { }
 
-    // named function expression
-    var named = function named (){
-      return true
-    }
+    // bad
+    function foo() { }
+
+    // good
+    const foo = function foo () { }
+    ```
 
+  - Function expressions:
+
+    ```javascript
     // immediately-invoked function expression (IIFE)
-    (function(){
-      console.log('Welcome to the Internet. Please follow me.')
+    (() => {
+      console.log('Welcome to the Internet. Please follow me.');
     })()
     ```
 
-  - Never declare a function in a non-function block (if, while, etc). Assign the function to a variable instead. Browsers will allow you to do it, but they all interpret it differently, which is bad news bears.
+  - Never declare a function in a non-function block (if, while, etc). Assign the function to a variable instead. Browsers will allow you to do it, but they all interpret it differently, which is bad news bears. It's also slower.
```

Comment:
Also, aren't variables declared with `const` and `let` block scoped? So they won't even work outside unless declared outside?